dags/gmail_data_import_pipeline.py

```diff
@@ -148,7 +150,63 @@ def gmail_link_message_thread_ids_etl(**kwargs):
             table_name=table_name,
             json_schema=generated_schema
         )
-        LOGGER.info('Created table: %s', table_name)
+
+        load_file_into_bq(
+            filename=filename,
+            dataset_name=data_config.dataset,
+            table_name=table_name,
+            project_name=data_config.project_name,
+            auto_detect_schema=True
+        )
+        LOGGER.info('Loaded table: %s', table_name)
+
+
+def gmail_thread_details_etl(**kwargs):
+    data_config = data_config_from_xcom(kwargs)
+    user_id = get_gmail_user_id()
+    table_name = data_config.table_name_thread_details
+
+    df_thread_id_list = get_distinct_values_from_bq(
+        project_name=data_config.project_name,
+        # will be changed !!!
+        dataset='hc_dev',  # data_config.dataset,
+        column_name=data_config.column_name_list_of_thread_ids,
+        table_name='test_gmail_ids'  # data_config.table_name_list_of_thread_ids
+        )
+
+    df_thread_details = pd.DataFrame()
+    df_thread_details_temp = pd.DataFrame()
+
+    for _, row in df_thread_id_list.iterrows():
+        df_thread_details_temp = get_one_thread(
+                                    get_gmail_service(),
+                                    user_id,
+                                    row[data_config.column_name_list_of_thread_ids]
+                                )
+        df_thread_details = pd.concat(
+                                [df_thread_details, df_thread_details_temp],
+                                ignore_index=True
+                            )
+
+    with TemporaryDirectory() as tmp_dir:
```

Comment:
It almost seems like this block appears three times. Maybe it would be worth refactoring.

Just mentioning it in case it's useful. In `data-science-dags` there is a [to_gbq method](https://github.com/elifesciences/data-science-dags/blob/46559fa41d8641137981d1490466e4e7363d0a3a/data_science_pipeline/utils/bq.py#L200) that is more or less the same as pandas' [to_gbq](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_gbq.html), except it uses our schema handling (well a copy of it since it's in another repo).

Reply:
Thank you for mentioning that, it can be considered for next version of this pipeline. For now, because of generating schema from the file it is easy to continue using files.